appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -1064,21 +1103,25 @@ Blockly.LexicalVariable.getEventParam = function (block) {
          } else {
            return null;
          }
-       } else if ( ( parent.type === "local_declaration_expression"
+       } else if ((type === "local_declaration_expression"
           && parent.getInputTargetBlock('RETURN') == child ) // only body is in scope of names
-          || ( parent.type === "local_declaration_statement"
+          || (type === "local_declaration_statement"
           && parent.getInputTargetBlock('STACK') == child ) // only body is in scope of names
            ) {
           var params = parent.getVars(); // [lyn, 10/13/13] Names from block, not localNames_ instance var
           if (params.indexOf(name) != -1) {
             return null; // Name is locally bound, not an event parameter.
           }
-       } else if ( ( (parent.type === "controls_forEach") || (parent.type === "controls_forRange") )
-                   && (parent.getInputTargetBlock('DO') == child) ) { // Only DO is in scope, not other inputs!
+       } else if ((type === "controls_forEach" || type === "controls_forRange")
+           && (parent.getInputTargetBlock('DO') == child) ) { // Only DO is in scope, not other inputs!
          var loopName = parent.getFieldValue('VAR');
-           if (loopName == name) {
-             return null; // Name is locally bound, not an event parameter.
+         if (loopName == name) {
+           return null; // Name is locally bound, not an event parameter.
          }
+       } else if (type == 'controls_for_each_dict'
+           && parent.getInputTargetBlock('DO') == child
+           && parent.getVars().indexOf(name) != -1) {
```

Comment:
I just want to confirm I understand why this check is here, which is that its purpose is to make sure that the key and value names don't collide. Is that correct? In either case, would you leave a comment for future developers why there's a check versus the other for constructs?

Reply:
So my understanding of this function is that it is used to check if a lexical get/set is accessing an event parameter, so that it can be tagged.

`parent.getInputTargetBlock('DO') == child`
Checks that the block is within the scope of vars defined by the for_each_dict block.

`parent.getVars()indexOf(name) != -1`
Checks that the variable being accessed is one of the vars defined by the for_each_dict block.

So if the `child` is within the scope && it is accessing a loop var, we return null to say that it is not accessing an event parameter.

Will add comments!